### PR TITLE
Region improvements, code legibility improvements

### PR DESCRIPTION
**Pull request by schmudde**
As the title says. I also retained the few code formatting tweaks suggested by @daveliepmann when he merged the previous PR.

### CODE REVIEW

File: src/appliedsciencestudio/covid19_clj_viz/italia.clj [hunk 1]
```diff
@@ -2,16 +2,19 @@
...
   `resources` directory."
   (:require [appliedsciencestudio.covid19-clj-viz.common :refer [oz-config]]
             [jsonista.core :as json]
             [meta-csv.core :as mcsv]
             [oz.core :as oz]))
 
+;;;;;;;;;;;;;;;;;;;;
+;; Conform Functions
+
+;; First, an Italian-to-English translation mapping for header names.
```
Comment (daveliepmann) [inline src/appliedsciencestudio/covid19_clj_viz/italia.clj:16]: Why not leave the docstring?

File: src/appliedsciencestudio/covid19_clj_viz/italia.clj [hunk 2]
```diff
@@ -51,51 +53,70 @@
...
+         (name-combined-region)                                ; name the combined region
+         (remove-inexact-data)                                 ; cleanup: remove longitude and latitude because it no longer applies
+         (conj region-covid-data))))                           ; add this back into the main collection
 
 (defn conform-to-territory-name
   "Index each map of territory information by territory name."
   [territories territory-key]
-  (into {} (map #(vector (territory-key %) %)
-                territories)))
+  (->> (map #(vector (territory-key %) %) territories)
```
Comment (daveliepmann) [inline src/appliedsciencestudio/covid19_clj_viz/italia.clj:74]: Purely for discussion: I tend to eliminate one-step thread macros. Do you prefer this style?

File: src/appliedsciencestudio/covid19_clj_viz/italia.clj [hunk 3]
```diff
@@ -51,51 +53,70 @@
...
+       (into {})))
 
-(defn compute-cases-per-100k [province-data-with-pop]
+(defn compute-cases-per-100k
```
Comment (daveliepmann) [inline src/appliedsciencestudio/covid19_clj_viz/italia.clj:77]: I may try to refactor this inline if I get a chance.

File: src/appliedsciencestudio/covid19_clj_viz/italia.clj [hunk 4]
```diff
@@ -51,51 +53,70 @@
...
-       (map #(update % :region-name (fn [region-name]
-                                      (if-let [update-region-name (normalize-region-names region-name)]
-                                        update-region-name
-                                        region-name))))
-       (normalize-trentino)))
+    (->> (map #(select-keys % keys-to-sum) regions-to-combine) ; grab the important keys from the regions we want to combine
+         (reduce #(merge-with + %1 %2))                        ; add the info from each key together
+         (conj (first regions-to-combine))                     ; add in the rest of the information (name, region number, etc...)
+         (name-combined-region)                                ; name the combined region
+         (remove-inexact-data)                                 ; cleanup: remove longitude and latitude because it no longer applies
```
Comment (daveliepmann) [inline src/appliedsciencestudio/covid19_clj_viz/italia.clj:68]: These `let`-bound functions would probably read better if inlined. You can keep the names. For example, just for `remove-inexact-data`:

```clojure
(->> (map #(select-keys % keys-to-sum) regions-to-combine)
         (reduce #(merge-with + %1 %2))
         (conj (first regions-to-combine))
         (name-combined-region)
         (fn remove-inexact-data [combined-regions] (dissoc combined-regions :lat :lon))
         (conj region-covid-data))
```

**Review by daveliepmann — CHANGES_REQUESTED** [review]

File: src/appliedsciencestudio/covid19_clj_viz/italia.clj (hunk 4 above)
Comment (schmudde) [reply src/appliedsciencestudio/covid19_clj_viz/italia.clj:68]: Great idea. This whole function is a bit embarrassing. I painted myself in a corner after not really thinking enough about how I'm going to conform the three data sources (covid, geography, and population).

File: src/appliedsciencestudio/covid19_clj_viz/italia.clj (hunk 2 above)
Comment (schmudde) [reply src/appliedsciencestudio/covid19_clj_viz/italia.clj:74]: I generally agree and I specifically agree in this case. I made an exception for 
```
(def region-population-data                                                                                                                                                                   
  "From http://www.comuni-italiani.it/province.html with updates to Trentino-Alto Adige/Südtirol and Valle d'Aosta/Vallée d'Aoste."
  (-> (mcsv/read-csv "resources/italy.region-population.csv" {:fields [:region-name :population :number-of-provinces]})                                                                       
      (conform-to-territory-name :region-name)))
```
The URL and all the fields seemed pretty gnarly to inline. This seemed to separate two very different processes.

**Pull request merged**